### PR TITLE
Fix Runaway Retries When HikariConnectionPool is Closed

### PR DESCRIPTION
**Pull request by bmarcaur**
**Goals (and why)**:
Some of our testing infrastructure for an internal project was experiencing a considerable amount of INFO logs due to `RetriableTransactions` quickly spinning when the underlying connection was closed. We are working to track down why the connection pool was closed, but the `AsyncPuncher` was not closed. However, since a closed connection pool throws a SQLException the `RetriableTransactions` believes that it needs to retry despite this scenario never recovering. This produced logs in excess of 5GB for a single test run.

**Implementation Description (bullets)**:

**Testing (What was existing testing like?  What have you done to improve it?)**:

**Concerns (what feedback would you like?)**:

**Where should we start reviewing?**:

**Priority (whenever / two weeks / yesterday)**:

### CODE REVIEW

File: commons-db/src/main/java/com/palantir/nexus/db/pool/RetriableTransactions.java [hunk 1]
```diff
@@ -171,6 +171,10 @@ public Throwable getError() {
                                     now,
                                     e);
                         }
+                        if (cm.isClosed()) {
+                            log.warn("Aborting tx retry, underlying connection is closed", e);
```
Comment (schlosna) [inline commons-db/src/main/java/com/palantir/nexus/db/pool/RetriableTransactions.java:175]: Nit as message should likely be an error and clarify the connection manager itself is closed

```suggestion
                            log.error("Aborting transaction retry, underlying connection manager is closed", e);
```

**Review by schlosna — COMMENTED** [review]
Overall looks good, one nit on the logging

**Comment by bmarcaur** [comment]
Thanks for the review @schlosna

**Comment by schlosna** [comment]
Hmm, something is up with CircleCI not running, may need to poke at that on Monday

**Review by sudiksha27 — APPROVED** [review]
LGTM!

**Review by schlosna — APPROVED** [review]

**Review by jeremyk-91 — APPROVED** [review]

**Pull request merged**